### PR TITLE
TestK8gbBasicRoundRobinExample

### LINKED ISSUE #389: Cover RoundRobin IP list merge with Terratest
Currently, we are checking actual DNS response contents at the e2e level only for Failover strategy
https://github.com/AbsaOSS/k8gb/blob/master/terratest/test/k8gb_basic_failover_test.go#L92

We need to extend RoundRobin tests with the similar approach checking that DNS response consists of IP addresses from both clusters

### PR DESCRIPTION
**Pull request by kuritka**
closes #389

TestK8gbBasicRoundRobinExample having four steps
 - round-robin on two concurrent clusters with podinfo running
 - kill podinfo on the first cluster
 - kill podinfo on the second cluster
 - start podinfo on the both clusters

Signed-off-by: kuritka <kuritka@gmail.com>

### CODE REVIEW

File: terratest/examples/roundrobin2.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: k8gb.absa.oss/v1beta1
+kind: Gslb
+metadata:
+  name: test-gslb
```
Comment (k0da) [inline terratest/examples/roundrobin2.yaml:4]: please choose a unique name

File: terratest/test/helpers.go [hunk 2]
```diff
@@ -185,3 +186,39 @@ func assertGslbDeleted(t *testing.T, options *k8s.KubectlOptions, gslbName strin
...
+func WaitForLocalGSLB(t *testing.T, host string, port int, expectedResult []string) (output []string, err error) {
+	const maxRetries = 100
+	const sleepBetweenRetries = time.Second * 1
+	var actionDescription = fmt.Sprintf("%s:%v", host, port)
+
+	if len(expectedResult) == 0 {
+		return
+	}
+	sort.Strings(expectedResult)
+	for i := 0; i <= maxRetries; i++ {
```
Comment (ytsarev) [inline terratest/test/helpers.go:201]: Conceptually and by implementation it is very close to existing https://github.com/AbsaOSS/k8gb/blob/2d1b7a6607631d1a75e4b6ff8d8a02c6d7feba8e/terratest/test/helpers.go#L68 . Is there any reason we are not reusing it?

File: terratest/examples/roundrobin2.yaml (hunk 1 above)
Comment (kuritka) [reply terratest/examples/roundrobin2.yaml]: amended unique name `roundrobin-test-gslb`.

File: terratest/test/k8gb_basic_roundrobin_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+// Basic k8gb deployment test that is verifying that associated ingress is getting created
+// Relies on two local clusters deployed by `$make deploy-two-local-clusters`
+func TestK8gbBasicRoundRobinExample(t *testing.T) {
+	t.Parallel()
+	const host = "roundrobin-test.cloud.example.com"
+	const gslbName = "roundrobin-test-gslb"
+	var waitForLocalGSLB = func(port int, expectedResult []string) (output []string, err error) {
```
Comment (ytsarev) [inline terratest/test/k8gb_basic_roundrobin_test.go:40]: Cool stuff on reusing existing helper. Still probably makes sense to put it to helpers as well so `waitForLocalGSLB` will be available in other tests?

Comment (kuritka) [reply terratest/test/k8gb_basic_roundrobin_test.go]: @ytsarev amended

File: terratest/test/k8gb_basic_roundrobin_test.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+	t.Run("round-robin on two concurrent clusters with podinfo running", func(t *testing.T) {
+		resolvedIPs5053, err := waitForLocalGSLB(t, host, 5053, expectedIPs)
+		require.NoError(t, err)
+		resolvedIPs5054, err := waitForLocalGSLB(t, host, 5054, expectedIPs)
+		require.NoError(t, err)
+
+		assert.NotEmpty(t, resolvedIPs5053)
+		assert.NotEmpty(t, resolvedIPs5054)
+		assert.Equal(t, len(resolvedIPs5053), len(resolvedIPs5053))
```
Comment (ytsarev) [inline terratest/test/k8gb_basic_roundrobin_test.go:81]:
```suggestion
		assert.Equal(t, len(resolvedIPs5053), len(expectedIPs))
```

File: terratest/test/k8gb_basic_roundrobin_test.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+		assertGslbStatus(t, optionsContext2, gslbName, host+":Healthy")
+
+		resolvedIPs5053, err := waitForLocalGSLB(t, host, 5053, expectedIPs)
+		require.NoError(t, err)
+		resolvedIPs5054, err := waitForLocalGSLB(t, host, 5054, expectedIPs)
+		require.NoError(t, err)
+
+		assert.NotEmpty(t, resolvedIPs5053)
+		assert.NotEmpty(t, resolvedIPs5054)
+		assert.Equal(t, len(resolvedIPs5053), len(resolvedIPs5053))
```
Comment (ytsarev) [inline terratest/test/k8gb_basic_roundrobin_test.go:126]:
```suggestion
		assert.Equal(t, len(resolvedIPs5053), len(expectedIPs))
```
or
```suggestion
		assert.Equal(t, len(resolvedIPs5053), len(resolvedIPs5054))
```
not sure, what was the original idea

**Review by ytsarev — CHANGES_REQUESTED** [review]

File: terratest/test/k8gb_basic_roundrobin_test.go (hunk 5 above)
Comment (kuritka) [reply terratest/test/k8gb_basic_roundrobin_test.go]: @ytsarev , thx for findings! fixed both occurrences

```go
		assert.Equal(t, len(resolvedIPs5053), len(expectedIPs))
		assert.Equal(t, len(resolvedIPs5054), len(expectedIPs))
```

**Review by ytsarev — APPROVED** [review]
Thanks, looks great

**Pull request merged**